c10/xpu/XPUFunctions.cpp

```diff
@@ -31,7 +31,8 @@ struct DevicePool {
   std::unique_ptr<sycl::context> context;
 } gDevicePool;
 
-void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
+// Ensures we only call enumDevices only once.
+void enumDevices() {
```

Comment:
This could maybe be moved into a lambda function guarded by `call_once` (see my other comment below).

Reply:
Change the parameter `devices` back to `enumDevices to prevent unintentional corrupting of the device list, since the `initGlobalDevicePoolState` only can be called once.